src/event_object_80688E4.c

```diff
@@ -0,0 +1,603 @@
+{
+    u8 animNum;
+    u8 spriteId = sub_806900C(objectEventId);
+    if (spriteId != MAX_SPRITES)
+    {
+        struct Sprite *sprite = &gSprites[spriteId];
+        StartSpriteAnim(sprite, ObjectEventDirectionToImageAnimId(direction));
+    }
+}
+
+void sub_8069094(u8 objectEventId, u8 direction)
+{
+    u8 animNum;
+    s32 spriteId = sub_806900C(objectEventId);
+    u16 baseBlock;
+    if (spriteId != MAX_SPRITES)
+    {
+        struct Sprite *sprite = &gSprites[spriteId];
+        const struct ObjectEventGraphicsInfo * info = GetObjectEventGraphicsInfo(direction);
+        baseBlock = sprite->oam.tileNum;
+        sprite->oam = *info->oam;
+        sprite->oam.tileNum = baseBlock;
+        sprite->oam.paletteNum = info->paletteSlot;
+        sprite->images = info->images;
+        if (info->subspriteTables == NULL)
+        {
+            sprite->subspriteTables = NULL;
+            sprite->subspriteTableNum = 0;
+            sprite->subspriteMode = 0;
+        }
+        else
+        {
+            SetSubspriteTables(sprite, info->subspriteTables);
+            sprite->subspriteMode = 2;
+        }
+        StartSpriteAnim(sprite, 0);
+    }
+}
+
+void sub_8069124(u8 objectEventId, bool32 arg1)
+{
+    u8 spriteId = sub_806900C(objectEventId);
+    if (spriteId != MAX_SPRITES)
+    {
+        if (arg1)
+            gSprites[spriteId].data[2] = TRUE;
+        else
+            gSprites[spriteId].data[2] = FALSE;
+    }
+}
+
+bool32 sub_806916C(u8 objectEventId)
+{
+    u8 spriteId = sub_806900C(objectEventId);
+    if (spriteId == MAX_SPRITES)
+        return FALSE;
+    return gSprites[spriteId].data[2] == TRUE;
+}
+
+void sub_80691A4(u8 objectEventId, u8 direction)
+{
+    u8 spriteId = sub_806900C(objectEventId);
+    if (spriteId != MAX_SPRITES)
+    {
+        gSprites[spriteId].data[3] = direction;
+        gSprites[spriteId].data[4] = 0;
+    }
+}
+
+void sub_80691D4(struct Sprite * sprite)
+{
+    switch (sprite->data[4])
+    {
+    case 0:
+        sprite->pos2.y = 0;
+        sprite->data[4]++;
+        // fallthrough
+    case 1:
+        if ((sprite->pos2.y -= 8) == -160)
+        {
+            sprite->pos2.y = 0;
+            sprite->data[2] = 1;
+            sprite->data[3] = 0;
+            sprite->data[4] = 0;
+        }
+    }
+}
+
+void sub_8069210(struct Sprite * sprite)
+{
+    switch (sprite->data[4])
+    {
+    case 0:
+        sprite->pos2.y = -160;
+        sprite->data[4]++;
+        // fallthrough
+    case 1:
+        if ((sprite->pos2.y += 8) == 0)
+        {
+            sprite->data[3] = 0;
+            sprite->data[4] = 0;
+        }
+    }
+}
+
+void sub_8069248(struct Sprite * sprite)
+{
+    switch (sprite->data[3])
+    {
+    case 0:
+        break;
+    case 1:
+        sub_8069210(sprite);
+        break;
+    case 2:
+        sub_80691D4(sprite);
+        break;
+    default:
+        sprite->data[3] = 0;
+        AGB_ASSERT_EX(0, ABSPATH("evobjmv.c"), 13331);
+    }
```

Comment:
add a `break` according to style guide (maybe also above 2 funcs)

Reply:
I'll put it onto the above two funcs but leave it off here because the intention is that there is no return from the AGB_ASSERT_EX.